pkg/cache/postgres.go

```diff
@@ -29,30 +31,40 @@ func NewPostgresCache(context context.Context) *postgresCache {
 	}
 }
 
+func (c *postgresCache) saveCheckAndStatus(check pkg.Check, status pkg.CheckStatus) (string, error) {
+	tx := c.DB().Begin()
```

Comment:
Why not do this for the entire list of checks ? a tx per check status is probably overkill